packages/docusaurus/src/client/clientEntry.tsx

```diff
@@ -30,7 +30,7 @@ if (ExecutionEnvironment.canUseDOM) {
   const renderMethod = process.env.NODE_ENV === 'production' ? hydrate : render;
   preload(routes, window.location.pathname).then(() => {
     renderMethod(
-      <BrowserRouter>
+      <BrowserRouter basename={window.DOCUSAURUS_ROUTER_BASE_NAME}>
```

Comment:
this would onyl work for the client-side (after React is loaded in the browser), but the server links wouldn't have the basename applied, unless you modify serverEntry.js `<StaticRouter location={location} context={context}>`, on which you can't use `window` because it's run by NodeJS.